pkg/agent/evictionmanager/plugin/memory/helper.go

```diff
@@ -0,0 +1,290 @@
+// if numa node is specified, return config in this numa; otherwise return system-level config
+func (e *EvictionHelper) getWatermarkMetrics(numaID int) (free, total, scaleFactor float64, err error) {
+	if numaID >= 0 {
+		free, err = e.metaServer.GetNumaMetric(numaID, consts.MetricMemFreeNuma)
+		if err != nil {
+			return 0, 0, 0, fmt.Errorf(errMsgGetNumaMetrics, consts.MetricMemFreeNuma, numaID, err)
+		}
+		_ = e.emitter.StoreFloat64(metricsNameNumaMetric, free, metrics.MetricTypeNameRaw,
+			metrics.ConvertMapToTags(map[string]string{
+				metricsTagKeyNumaID:     strconv.Itoa(numaID),
+				metricsTagKeyMetricName: consts.MetricMemFreeNuma,
+			})...)
+
+		total, err = e.metaServer.GetNumaMetric(numaID, consts.MetricMemTotalNuma)
+		if err != nil {
+			return 0, 0, 0, fmt.Errorf(errMsgGetNumaMetrics, consts.MetricMemTotalNuma, numaID, err)
+		}
+		_ = e.emitter.StoreFloat64(metricsNameNumaMetric, total, metrics.MetricTypeNameRaw,
+			metrics.ConvertMapToTags(map[string]string{
+				metricsTagKeyNumaID:     strconv.Itoa(numaID),
+				metricsTagKeyMetricName: consts.MetricMemTotalNuma,
+			})...)
+	} else {
+		free, err = e.metaServer.GetNodeMetric(consts.MetricMemFreeSystem)
+		if err != nil {
+			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemFreeSystem, err)
+		}
+		_ = e.emitter.StoreFloat64(metricsNameSystemMetric, free, metrics.MetricTypeNameRaw,
+			metrics.ConvertMapToTags(map[string]string{
+				metricsTagKeyMetricName: consts.MetricMemFreeSystem,
+			})...)
+
+		total, err = e.metaServer.GetNodeMetric(consts.MetricMemTotalSystem)
+		if err != nil {
+			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemTotalSystem, err)
+		}
+		_ = e.emitter.StoreFloat64(metricsNameSystemMetric, total, metrics.MetricTypeNameRaw,
+			metrics.ConvertMapToTags(map[string]string{
+				metricsTagKeyMetricName: consts.MetricMemTotalSystem,
+			})...)
+	}
+
+	scaleFactor, err = e.metaServer.GetNodeMetric(consts.MetricMemScaleFactorSystem)
+	if err != nil {
+		return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemScaleFactorSystem, err)
+	}
+	_ = e.emitter.StoreFloat64(metricsNameSystemMetric, scaleFactor, metrics.MetricTypeNameRaw,
+		metrics.ConvertMapToTags(map[string]string{
+			metricsTagKeyMetricName: consts.MetricMemScaleFactorSystem,
+		})...)
+
+	return free, total, scaleFactor, nil
+}
+
+func (e *EvictionHelper) getSystemKswapdStealMetrics() (float64, error) {
+	kswapdSteal, err := e.metaServer.GetNodeMetric(consts.MetricMemKswapdstealSystem)
+	if err != nil {
+		return 0, fmt.Errorf(errMsgGetSystemMetrics, "mem.kswapdsteal.system", err)
+	}
+	_ = e.emitter.StoreFloat64(metricsNameSystemMetric, kswapdSteal, metrics.MetricTypeNameRaw,
+		metrics.ConvertMapToTags(map[string]string{
+			metricsTagKeyMetricName: consts.MetricMemKswapdstealSystem,
+		})...)
+
+	return kswapdSteal, nil
+}
+
+func (e *EvictionHelper) selectPodsToEvict(activePods []*v1.Pod, topN uint64, numaID,
```

Comment:
since this logic if only used by system and numa plugin, and rss doesn't follow this. should we rename the function to clarify?

Reply:
yeah, current name might be misleading.I'll rename it but I think the name should more tied with its functionality rather than with current callers. Maybe other caller will call it someday.